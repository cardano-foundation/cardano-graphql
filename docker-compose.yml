version: "3.5"

services:
  postgres:
    image: postgres:${POSTGRES_VERSION:-11.5-alpine}
    environment:
      - POSTGRES_LOGGING=true
      - POSTGRES_DB_FILE=/run/secrets/postgres_db
      - POSTGRES_PASSWORD_FILE=/run/secrets/postgres_password
      - POSTGRES_USER_FILE=/run/secrets/postgres_user
    secrets:
      - postgres_db
      - postgres_password
      - postgres_user
    shm_size: '2gb'
    volumes:
      - postgres-data:/var/lib/postgresql/data
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
  cardano-node:
    image: inputoutput/cardano-node:${CARDANO_NODE_VERSION:-1.13.0}
    environment:
      - NETWORK=${NETWORK:-mainnet}
    volumes:
      - node-db:/data/db
      - node-ipc:/ipc
    logging:
      driver: "json-file"
      options:
        max-size: "400k"
        max-file: "20"
  cardano-db-sync-extended:
    image: inputoutput/cardano-db-sync:${CARDANO_DB_SYNC_VERSION:-2.1.0}
    environment:
      - EXTENDED=true
      - NETWORK=${NETWORK:-mainnet}
      - POSTGRES_HOST=postgres
      - POSTGRES_PORT=5432
    depends_on:
      - cardano-node
      - postgres
    secrets:
      - postgres_password
      - postgres_user
      - postgres_db
    volumes:
      - node-ipc:/node-ipc
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
  hasura:
    build:
      context: ./packages/api-cardano-db-hasura/hasura
    image: inputoutput/cardano-graphql-hasura:${CARDANO_GRAPHQL_VERSION:-v1.0.0-rc.13}
    ports:
      - "8090:8080"
    depends_on:
      - "postgres"
    restart: on-failure
    environment:
      - POSTGRES_HOST=postgres
      - POSTGRES_PORT=5432
      - POSTGRES_DB_FILE=/run/secrets/postgres_db
      - POSTGRES_PASSWORD_FILE=/run/secrets/postgres_password
      - POSTGRES_USER_FILE=/run/secrets/postgres_user
      - HASURA_GRAPHQL_ENABLE_CONSOLE=true
      - HASURA_GRAPHQL_ENABLED_LOG_TYPES=startup, http-log, webhook-log, websocket-log, query-log
      - HASURA_GRAPHQL_ENABLE_TELEMETRY=false
    secrets:
      - postgres_db
      - postgres_password
      - postgres_user
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
  cardano-graphql:
    build:
      context: .
      target: server
    image: inputoutput/cardano-graphql:${CARDANO_GRAPHQL_VERSION:-v1.0.0-rc.13}
    environment:
      - CACHE_ENABLED=true
      - HASURA_URI=http://hasura:8080
    expose:
      - 3100
    ports:
      - ${API_PORT:-3100}:3100
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"
secrets:
  postgres_db:
    file: ./config/secrets/postgres_db
  postgres_password:
    file: ./config/secrets/postgres_password
  postgres_user:
    file: ./config/secrets/postgres_user
volumes:
  node-db:
  node-ipc:
  postgres-data:
