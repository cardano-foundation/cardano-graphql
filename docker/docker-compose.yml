version: '3.7'
services:
  api-server:
    build:
      context: ..
      target: server
    environment:
      - API_PORT=3000
    ports:
      - 4000:3000
