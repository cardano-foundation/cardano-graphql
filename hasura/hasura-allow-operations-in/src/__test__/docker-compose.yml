version: "3.5"

services:
  postgres:
    image: postgres:11.5-alpine
    environment:
      - POSTGRES_DB=test
      - POSTGRES_PASSWORD=password
      - POSTGRES_USER=postgres
  hasura:
    image: hasura/graphql-engine:v1.1.1
    ports:
      - "8091:8080"
    depends_on:
      - "postgres"
    restart: on-failure
    environment:
      - HASURA_GRAPHQL_DATABASE_URL=postgres://postgres:password@postgres:5432/test
      - HASURA_GRAPHQL_ENABLE_CONSOLE=true
      - HASURA_GRAPHQL_ENABLED_LOG_TYPES=startup, http-log, query-log
