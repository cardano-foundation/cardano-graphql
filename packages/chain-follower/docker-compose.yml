version: "3.5"

services:
  rabbitmq:
    image: rabbitmq:3-management
    ports:
        - 5672:5672
        - 15672:15672
    volumes:
        - rabbitmq:/var/lib/rabbitmq/
        
  dgraph:
    image: dgraph/standalone:v21.03.0
    ports:
      - 5080:5080
      - 6080:6080
      - 8080:8080
      - 9080:9080
      - 8000:8000
    volumes:
      - dgraph:/dgraph

  cardano-node-ogmios:
    build:
      args:
        - NETWORK=${NETWORK:-mainnet}
      cache_from: [cardanosolutions/cardano-node-ogmios:latest]
      context: ../../ogmios
    image: cardanosolutions/cardano-node-ogmios:${CARDANO_NODE_OGMIOS_VERSION:-v4.1.0}-${NETWORK:-mainnet}
    logging:
      driver: "json-file"
      options:
        max-size: "400k"
        max-file: "20"
    ports:
      - ${OGMIOS_PORT:-1337}:1337
    restart: on-failure
    volumes:
      - node-db:/db

  chain-follower:
    build:
      args:
        - NETWORK=${NETWORK:-mainnet}
      cache_from: [inputoutput/cardano-graphql-chain-follower:latest]
      context: ../../
      target: chain-follower
    image: inputoutput/cardano-graphql-chain-follower:${CARDANO_GRAPHQL_VERSION:-6.0.0}-${NETWORK:-mainnet}
    environment:
      - LOGGER_MIN_SEVERITY=${LOGGER_MIN_SEVERITY:-info}
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-size: "200k"
        max-file: "10"

volumes:
  node-db:
  dgraph:
  rabbitmq:
