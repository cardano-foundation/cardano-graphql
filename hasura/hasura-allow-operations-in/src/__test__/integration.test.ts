import * as path from 'path';
import * as compose from 'docker-compose';
import { IDockerComposeOptions } from 'docker-compose';
import * as delay from 'delay';
import { run } from '../';

const hasuraUri = process.env.HASURA_URI || 'http://localhost:8091';
const sourcePathOne = path.resolve(__dirname, 'operations/1/**/*.graphql');
const sourcePathTwo = path.resolve(__dirname, 'operations/2/**/*.graphql');
const composeOptions: IDockerComposeOptions['composeOptions'] = [
  '-p hasura-allow-operations-in',
];

describe('e2e', () => {
  afterEach(async () => {
    await compose.down({
      composeOptions,
      cwd: path.join(__dirname),
      log: true,
    });
  });

  describe('clean state', () => {
    beforeEach(async () => {
      await compose.upAll({
        cwd: path.join(__dirname),
        composeOptions,
        commandOptions: ['--force-recreate'],
        log: true,
      });
      await delay(3000);
    });

    it('finds GraphQL operations as SDL within .graphql files, and adds them to the Hasura allow list', async () => {
      const report = await run(hasuraUri, sourcePathOne);
      expect(report.existingCount).toBe(0);
      expect(report.addedCount).toBe(2);
      expect(JSON.stringify(report, null, 4)).toMatchSnapshot();
    });

    it('optionally includes the introspection query', async () => {
      const report = await run(hasuraUri, sourcePathOne, true);
      expect(report.existingCount).toBe(0);
      expect(report.addedCount).toBe(3);
      expect(JSON.stringify(report, null, 4)).toMatchSnapshot();
    });
  });

  describe('previous state', () => {
    beforeEach(async () => {
      await compose.upAll({
        cwd: path.join(__dirname),
        composeOptions,
        log: true,
      });
      await delay(2000);
      const report = await run(hasuraUri, sourcePathOne);
      expect(report.addedCount).toBe(2);
    });

    it('finds GraphQL operations as SDL within .graphql files, and adds new items to the Hasura allow list', async () => {
      const report = await run(hasuraUri, sourcePathOne);
      expect(report.existingCount).toBe(2);
      expect(report.collectionCreated).toBe(false);
      expect(report.addedCount).toBe(0);
      expect(JSON.stringify(report, null, 4)).toMatchSnapshot();
    });

    it('allows new queries to be added from different source pointers', async () => {
      const report = await run(hasuraUri, sourcePathTwo, true);
      expect(report.existingCount).toBe(0);
      expect(report.collectionCreated).toBe(false);
      expect(report.addedCount).toBe(2);
      expect(JSON.stringify(report, null, 4)).toMatchSnapshot();
    });
  });

  describe('Hasura unavailable', () => {
    it('throws if the endpoint cannot be reached', async () => {
      expect.assertions(1);
      try {
        await run(hasuraUri, sourcePathOne);
      } catch (error) {
        expect(error).toMatchSnapshot();
      }
    });
  });
});
