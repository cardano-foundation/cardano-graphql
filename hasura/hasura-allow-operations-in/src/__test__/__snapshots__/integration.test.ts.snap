// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e Hasura unavailable throws if the endpoint cannot be reached 1`] = `[Error: Network Error]`;

exports[`e2e clean state finds GraphQL operations as SDL within .graphql files, and adds them to the Hasura allow list 1`] = `
"{
    \\"addedCount\\": 2,
    \\"existingCount\\": 0,
    \\"collectionCreated\\": true,
    \\"operationDefinitionsFound\\": [
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"firstQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 16
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"three\\",
                            \\"loc\\": {
                                \\"start\\": 21,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"four\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 37
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 37
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 41
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 21,
                            \\"end\\": 41
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 17,
                    \\"end\\": 43
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 43
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"secondQuery\\",
                \\"loc\\": {
                    \\"start\\": 51,
                    \\"end\\": 62
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"five\\",
                            \\"loc\\": {
                                \\"start\\": 67,
                                \\"end\\": 71
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"six\\",
                                        \\"loc\\": {
                                            \\"start\\": 78,
                                            \\"end\\": 81
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 78,
                                        \\"end\\": 81
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 72,
                                \\"end\\": 85
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 67,
                            \\"end\\": 85
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 63,
                    \\"end\\": 87
                }
            },
            \\"loc\\": {
                \\"start\\": 45,
                \\"end\\": 87
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"aSingleQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 18
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"one\\",
                            \\"loc\\": {
                                \\"start\\": 23,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"two\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 36
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 36
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 40
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 23,
                            \\"end\\": 40
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 19,
                    \\"end\\": 42
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 42
            }
        }
    ]
}"
`;

exports[`e2e clean state optionally includes the introspection query 1`] = `
"{
    \\"addedCount\\": 3,
    \\"existingCount\\": 0,
    \\"collectionCreated\\": true,
    \\"introspectionAllowed\\": true,
    \\"operationDefinitionsFound\\": [
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"firstQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 16
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"three\\",
                            \\"loc\\": {
                                \\"start\\": 21,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"four\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 37
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 37
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 41
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 21,
                            \\"end\\": 41
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 17,
                    \\"end\\": 43
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 43
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"secondQuery\\",
                \\"loc\\": {
                    \\"start\\": 51,
                    \\"end\\": 62
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"five\\",
                            \\"loc\\": {
                                \\"start\\": 67,
                                \\"end\\": 71
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"six\\",
                                        \\"loc\\": {
                                            \\"start\\": 78,
                                            \\"end\\": 81
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 78,
                                        \\"end\\": 81
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 72,
                                \\"end\\": 85
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 67,
                            \\"end\\": 85
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 63,
                    \\"end\\": 87
                }
            },
            \\"loc\\": {
                \\"start\\": 45,
                \\"end\\": 87
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"aSingleQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 18
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"one\\",
                            \\"loc\\": {
                                \\"start\\": 23,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"two\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 36
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 36
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 40
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 23,
                            \\"end\\": 40
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 19,
                    \\"end\\": 42
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 42
            }
        }
    ]
}"
`;

exports[`e2e previous state allows new queries to be added from different source pointers 1`] = `
"{
    \\"addedCount\\": 2,
    \\"existingCount\\": 0,
    \\"collectionCreated\\": false,
    \\"introspectionAllowed\\": true,
    \\"operationDefinitionsFound\\": [
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"another\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 13
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"seven\\",
                            \\"loc\\": {
                                \\"start\\": 18,
                                \\"end\\": 23
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"eight\\",
                                        \\"loc\\": {
                                            \\"start\\": 30,
                                            \\"end\\": 35
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 30,
                                        \\"end\\": 35
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 24,
                                \\"end\\": 39
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 18,
                            \\"end\\": 39
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 14,
                    \\"end\\": 41
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 41
            }
        }
    ]
}"
`;

exports[`e2e previous state finds GraphQL operations as SDL within .graphql files, and adds new items to the Hasura allow list 1`] = `
"{
    \\"addedCount\\": 0,
    \\"existingCount\\": 2,
    \\"collectionCreated\\": false,
    \\"operationDefinitionsFound\\": [
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"firstQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 16
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"three\\",
                            \\"loc\\": {
                                \\"start\\": 21,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"four\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 37
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 37
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 41
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 21,
                            \\"end\\": 41
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 17,
                    \\"end\\": 43
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 43
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"secondQuery\\",
                \\"loc\\": {
                    \\"start\\": 51,
                    \\"end\\": 62
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"five\\",
                            \\"loc\\": {
                                \\"start\\": 67,
                                \\"end\\": 71
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"six\\",
                                        \\"loc\\": {
                                            \\"start\\": 78,
                                            \\"end\\": 81
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 78,
                                        \\"end\\": 81
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 72,
                                \\"end\\": 85
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 67,
                            \\"end\\": 85
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 63,
                    \\"end\\": 87
                }
            },
            \\"loc\\": {
                \\"start\\": 45,
                \\"end\\": 87
            }
        },
        {
            \\"kind\\": \\"OperationDefinition\\",
            \\"operation\\": \\"query\\",
            \\"name\\": {
                \\"kind\\": \\"Name\\",
                \\"value\\": \\"aSingleQuery\\",
                \\"loc\\": {
                    \\"start\\": 6,
                    \\"end\\": 18
                }
            },
            \\"variableDefinitions\\": [],
            \\"directives\\": [],
            \\"selectionSet\\": {
                \\"kind\\": \\"SelectionSet\\",
                \\"selections\\": [
                    {
                        \\"kind\\": \\"Field\\",
                        \\"name\\": {
                            \\"kind\\": \\"Name\\",
                            \\"value\\": \\"one\\",
                            \\"loc\\": {
                                \\"start\\": 23,
                                \\"end\\": 26
                            }
                        },
                        \\"arguments\\": [],
                        \\"directives\\": [],
                        \\"selectionSet\\": {
                            \\"kind\\": \\"SelectionSet\\",
                            \\"selections\\": [
                                {
                                    \\"kind\\": \\"Field\\",
                                    \\"name\\": {
                                        \\"kind\\": \\"Name\\",
                                        \\"value\\": \\"two\\",
                                        \\"loc\\": {
                                            \\"start\\": 33,
                                            \\"end\\": 36
                                        }
                                    },
                                    \\"arguments\\": [],
                                    \\"directives\\": [],
                                    \\"loc\\": {
                                        \\"start\\": 33,
                                        \\"end\\": 36
                                    }
                                }
                            ],
                            \\"loc\\": {
                                \\"start\\": 27,
                                \\"end\\": 40
                            }
                        },
                        \\"loc\\": {
                            \\"start\\": 23,
                            \\"end\\": 40
                        }
                    }
                ],
                \\"loc\\": {
                    \\"start\\": 19,
                    \\"end\\": 42
                }
            },
            \\"loc\\": {
                \\"start\\": 0,
                \\"end\\": 42
            }
        }
    ]
}"
`;
